test/expiry.test.ts

```diff
@@ -137,3 +137,51 @@ test('returns cookies before the max age', async () => {
     ['cookieName', { expires, name: 'cookieName', maxAge: 2, value: 'abc-123' }],
   ])
 })
+
+test('does overwrite exisiting cookies which would not expire', () => {
+  const req = new Request('https://mswjs.io')
+
+  store.add(req, new Response(null, {
+    headers: new Headers({ 'set-cookie': `cookieName=abc-123 }` }),
+  }))
+
+  const date = new Date()
+  date.setSeconds(date.getSeconds() - 2)
+  const res = new Response(null, {
+    headers: new Headers({ 'set-cookie': `cookieName=abc-123; Expires=${ date.toUTCString() }` }),
+  })
+
+  store.add(req, res)
+
+  // Assert cookie entry has been deleted.
+  const allCookies = store.getAll()
```

Comment:
The `.getAll()` method returns the internal `store`, which is a Map, so you can access its `size` property instead of converting it to a list for that sake:

```ts
const allCookies = store.getAll()
expect(allCookies.size).toBe(0)
```